onboard/catkin_ws/src/task_planning/scripts/combination_tasks.py

```diff
@@ -0,0 +1,125 @@
+
+from task import Task
+
+
+class ListTask(Task):
+    """Run a list of tasks sequentially"""
+
+    def __init__(self, tasks, num_loops=1):
+        super(ListTask, self).__init__()
+
+        self.num_loops = num_loops
+        self.tasks = tasks
+        self.curr_index = 0
+
+    def _on_task_run(self):
+        if self.curr_index == len(self.tasks):
+            if self.num_loops > 0:
+                self.num_loops -= 1
+                if self.num_loops == 0:
+                    self.finish()
```

Comment:
reduce the nesting/complexity of this code.

Reply:
There isn't anything we can do to simplify this, since num_loops being <= 0 represents an infinite loop and is a special case.